imagefit/conf.py

```diff
@@ -35,8 +35,12 @@ class ImagefitConf(AppConf):
     }
 
     #: ConditionalGetMiddleware is required for browser caching
-    if not 'django.middleware.http.ConditionalGetMiddleware' in settings.MIDDLEWARE_CLASSES:
-        settings.MIDDLEWARE_CLASSES += ('django.middleware.http.ConditionalGetMiddleware',)
+    # Django's middleware management (and some other things) has changed
+    # since version 1.9.
+    property_name = (hasattr(settings, 'MIDDLEWARE') and 'MIDDLEWARE') or 'MIDDLEWARE_CLASSES'
+    middleswares = getattr(settings, property_name)
+    if not 'django.middleware.http.ConditionalGetMiddleware' in property_name:
```

Comment:
This can never be `True` as `property_name` is a _string_.
It should more be like:

```python
if 'django.middleware.http.ConditionalGetMiddleware' not in middlewares:
```

Reply:
In fact, I didn't touched this, it was already in the code ([see here](https://github.com/vinyll/django-imagefit/blob/748afbad1305cc6cc87602dcc891f7045641b050/imagefit/conf.py#L38)) ([and it works](https://stackoverflow.com/a/3481700)) but I can change it if you want. ;)

PS : Sorry for the delays, I'm on CEST timezone.

Reply:
Mhh weird, I replied to this but it seems it never got published… sorry for delay too then :/

The main issue I'm exposing here is that `property_name` is a _string_; you want to check if `'…ConditionalGetMiddleware'` is in the `middlewares` _tuple_ instead. Hence checking _django.middleware.http.ConditionalGetMiddleware_ in the _middles_ tuple and not in the _property_name_ string ;)

Side note as you exposed is that pep8 recommends writing `if … not in …`: https://pep8.org/#programming-recommendations.
I know the current code is doing it the other way and it works, but I wasn't aware of that pep8 point 5 years ago 👵